01-data-structures/02-stacks-and-queues/myqueue/myqueue.rb

```diff
@@ -5,14 +5,44 @@ class MyQueue
   def initialize
     @queue = Array.new
     @head = @queue[0]
+    @tail = @queue[-1]
   end
 
   def enqueue(element)
+    new_array = []
+    new_array[0] = element
+
+    @queue.each do |element|
+      new_array << element
+    end
+
+    @queue = new_array
+
+    @head = @queue[0]
+    @tail = @queue[-1]
   end
 
   def dequeue
+    # @queue.delete_at(-1)
```

Comment:
It is not wrong for you to use the `delete_at` method. Knowing that and knowing the semantics you have applied to which end of the array you append values and which end you remove them, you could make this method more concise without breaking any rules. However, this solution works for me.